src/app/home/pages/TokensPage/TokensPage.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import {
+  Card,
+  PageSection,
+  TextContent,
+  Text,
+  CardBody,
+  EmptyState,
+  EmptyStateIcon,
+  Title,
+} from '@patternfly/react-core';
+import { WrenchIcon } from '@patternfly/react-icons';
+import spacing from '@patternfly/react-styles/css/utilities/Spacing/spacing';
+import clusterSelectors from '../../../cluster/duck/selectors';
+import TokensTable from './components/TokensTable';
+import { useOpenModal } from '../../duck/hooks';
+import { IReduxState } from '../../../../reducers';
+import { IToken } from '../../../token/duck/types';
+import { ICluster } from '../../../cluster/duck/types';
+
+interface ITokensPageBaseProps {
+  tokenList: IToken[];
+  clusterList: ICluster[];
+}
+
+const TokensPageBase: React.FunctionComponent<ITokensPageBaseProps> = ({
+  tokenList,
+  clusterList,
+}: ITokensPageBaseProps) => {
+  const [isAddEditModalOpen, reallyToggleAddEditModal] = useOpenModal(false);
+  const toggleAddEditModal = () => {
+    alert('NATODO: modal still being implemented!');
+    reallyToggleAddEditModal();
+  };
+
+  return (
+    <>
+      <PageSection variant="light">
+        <TextContent>
+          <Text component="h1" className={spacing.mbAuto}>
+            Tokens
+          </Text>
+        </TextContent>
+      </PageSection>
+      <PageSection>
+        <Card>
+          <CardBody>
+            {/* NATODO add a TokenContext provider here when that becomes necessary? */}
```

Comment:
IIRC that was how we were delivering our polling triggers? We'll definitely want to be polling for token status.

Reply:
Yeah exactly, that's how we pass down `watchAddEditStatus` in the other pages. I didn't add it yet just because I didn't want to wire up an empty noop context before we actually get that polling implemented.